bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -0,0 +1,498 @@
+            throw new IllegalStateException("pollTask should be unregistered before registering a new one!");
+        }
+        if (!config.isPresent() || !endpoint.isPresent()) {
+            throw new IllegalStateException("registerPollTask called without proper configuration");
+        }
+
+        logger.debug("Setting up regular polling");
+
+        BasicModbusReadRequestBlueprint request = new BasicModbusReadRequestBlueprint(getSlaveId(),
+                ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, mainBlock.address, mainBlock.length,
+                config.get().maxTries);
+
+        pollTask = new BasicPollTaskImpl(endpoint.get(), request, new ModbusReadCallback() {
+
+            @Override
+            public void onRegisters(@Nullable ModbusReadRequestBlueprint request,
+                    @Nullable ModbusRegisterArray registers) {
+                if (registers == null) {
+                    logger.info("Received empty register array on poll");
+                    return;
+                }
+
+                handlePolledData(registers);
+
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }
+
+            @Override
+            public void onError(@Nullable ModbusReadRequestBlueprint request, @Nullable Exception error) {
+                handleError(error);
+            }
+
+            @Override
+            public void onBits(@Nullable ModbusReadRequestBlueprint request, @Nullable BitArray bits) {
+                // don't care, we don't expect this result
+            }
+        });
+
+        long refreshMillis = config.get().getRefreshMillis();
+        if (pollTask != null) {
+            PollTask task = pollTask;
+            managerRef.registerRegularPoll(task, refreshMillis, 1000);
+        }
+    }
+
+    /**
+     * This method should handle incoming poll data, and update the channels
+     * with the values received
+     */
+    protected abstract void handlePolledData(ModbusRegisterArray registers);
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+
+        logger.debug("Thing status changed to {}", this.getThing().getStatus().name());
+        if (getThing().getStatus() == ThingStatus.ONLINE) {
+            startUp();
+        } else if (getThing().getStatus() == ThingStatus.OFFLINE) {
+            tearDown();
+        }
+    }
+
+    /**
+     * Unregister poll task.
+     *
+     * No-op in case no poll task is registered, or if the initialization is incomplete.
+     */
+    private synchronized void unregisterPollTask() {
+        if (pollTask == null) {
+            return;
+        }
+        logger.debug("Unregistering polling from ModbusManager");
+        PollTask task = (@NonNull PollTask) pollTask;
+        managerRef.unregisterRegularPoll(task);
+
+        pollTask = null;
```

Comment:
This should work, if not let me know and I'll checkout your branch and resolve it myself.
```suggestion
        @Nullable PollTask task = pollTask;
        if(task == null){
            return;
        }
        logger.debug("Unregistering polling from ModbusManager");
        managerRef.unregisterRegularPoll(task);

        pollTask = null;
```

Reply:
Thank you @cpmeister I get it.
Changed the optional attributes in this class to @Nullable types.